lldb/source/Plugins/SymbolFile/DWARF/DWARFDIE.cpp

```diff
@@ -440,12 +440,6 @@ static void GetTypeLookupContextImpl(DWARFDIE die,
       continue;
     }
 
-    // If there is no name, then there is no need to look anything up for this
-    // DIE.
-    const char *name = die.GetName();
-    if (!name || !name[0])
-      return;
-
```

Comment:
Do we want to only pull out this match if this is a `DW_TAG_namespace`? Or is it useful to have unnamed structs/unions/classes in the contenxt?

Reply:
unnamed structs cannot contain [named structs](https://godbolt.org/z/sEfdMEvsz), so the only place these can appear is at the top (bottom?) level. We don't currently have a way to look them up from our APIs taking strings (we'd need something like an `(anonymous struct)`), but they could be looked up internally by code which constructs CompilerContexts directly. So yeah, I think including them here is the right choice for this function.